static/app/components/guidedSteps/guidedSteps.stories.tsx

```diff
@@ -0,0 +1,110 @@
+        <GuidedSteps>
+          <GuidedSteps.Step title="Step 1 Title">
+            This is the first step.
+            <GuidedSteps.StepButtons />
+          </GuidedSteps.Step>
+          <GuidedSteps.Step title="Step 2 Title">
+            This is the second step.
+            <GuidedSteps.StepButtons />
+          </GuidedSteps.Step>
+          <GuidedSteps.Step title="Step 3 Title">
+            This is the third step.
+            <GuidedSteps.StepButtons />
+          </GuidedSteps.Step>
+        </GuidedSteps>
+      </SizingWindow>
+    </Fragment>
+  ));
+
+  story('Custom button behavior', () => {
+    function SkipToLastButton() {
+      const {setCurrentStep, totalSteps} = useGuidedStepsContext();
+      return (
+        <GuidedSteps.ButtonWrapper>
+          <Button size="sm" onClick={() => setCurrentStep(totalSteps)}>
+            Skip to Last Step
+          </Button>
+        </GuidedSteps.ButtonWrapper>
+      );
+    }
+
+    return (
+      <Fragment>
+        <p>
+          A hook is provided to access and control the step state:{' '}
+          <code>useGuidedStepsContext()</code>. This can be used to create step buttons
+          with custom behavior.
+        </p>
+        <SizingWindow display="block">
+          <GuidedSteps>
+            <GuidedSteps.Step title="Step 1 Title">
+              This is the first step.
+              <SkipToLastButton />
+            </GuidedSteps.Step>
+            <GuidedSteps.Step title="Step 2 Title">
+              This is the second step.
+              <GuidedSteps.StepButtons />
+            </GuidedSteps.Step>
+            <GuidedSteps.Step title="Step 3 Title">
+              This is the third step.
+              <GuidedSteps.StepButtons />
+            </GuidedSteps.Step>
+          </GuidedSteps>
+        </SizingWindow>
+      </Fragment>
+    );
+  });
+
+  story('Controlling completed state', () => {
+    return (
+      <Fragment>
+        <p>
+          By default, previous steps are considered completed. However, if the completed
+          state is known it can be controlled using the <code>isCompleted</code> property
+          on <JSXNode name="GuidedSteps.Step" />. The GuidedStep component will begin on
+          the first incomplete step.
+        </p>
+        <SizingWindow display="block">
+          <GuidedSteps>
+            <GuidedSteps.Step title="Step 1 Title" isCompleted>
+              Congrats, you finished the first step!
+              <GuidedSteps.StepButtons />
+            </GuidedSteps.Step>
+            <GuidedSteps.Step title="Step 2 Title" isCompleted={false}>
+              You haven't completed the second step yet, here's how you do it.
+              <GuidedSteps.StepButtons />
+            </GuidedSteps.Step>
+            <GuidedSteps.Step title="Step 3 Title" isCompleted={false}>
+              You haven't completed the third step yet, here's how you do it.
+              <GuidedSteps.StepButtons />
+            </GuidedSteps.Step>
```

Comment:
Could `<GuidedSteps>` take a callback param that's triggered when we change which step is complete/visible?

In this case where `isCompleted` is controlled, we can't know when to add or remove checkboxes as the user moves through the steps. So the user could be clicking next next next, but nothing is getting checked off.
A callback could also be useful for saving the state of the checklist somewhere else, so even if the `<Step>` is not controlled we can emit analytics or something as they progress.

Reply:
Good callout! Does seem useful, will add

Reply:
It might also make sense, api design wise, to separate 'isComplete' states from 'isVisible', but have the default mode be to keep them in lock step with each other. Right now the "Controlling completed state" story allows you to see the 3rd step, while the 2nd step is incomplete. But it seems more like a buggy case based on the descriptions.

Reply:
So far, based on the two examples we have, we want the user to be able to skip ahead even if they haven't completed all previous ones. But I can see how we might want to lock the user into a step until it has been completed.

Doing this _should_ be possible with the current API though. You would simply not render a next button until the step is completed. So I'm not sure if we need another prop to do this?

Reply:
> we want the user to be able to skip ahead even if they haven't completed all previous ones

This is the current logic:
```
const isActive = currentStep === stepNumber;
const isCompleted = completedOverride ?? currentStep > stepNumber;
```
I'm not sure how people would skip over steps with these conditions being in the Step render method.

Reply:
By "skip ahead", I just mean that they are are able to go on to next/previous steps regardless of whether they are completed or not. `isActive` means that the step is currently being shown and all `isCompleted` does is display the checkmark.